Kahi_impactu_postcalculations/kahi_impactu_postcalculations/Kahi_impactu_postcalculations.py

```diff
@@ -114,26 +123,41 @@ def run(self):
                         self.author_count
                     ) for oaid in authors_ids)
 
-        # Getting the top words for each institution
-        print("INFO: Creating top words for institutions")
+        # Getting the top words for institution - step 1
+        if self.verbose > 1:
+            print("INFO: Creating top words for institutions - step 1")
         Parallel(
             n_jobs=self.n_jobs,
             verbose=10,
             backend="multiprocessing")(
                 delayed(top_words)(
-                    "affiliations",
-                    aff,
-                    "authors.affiliations.id"
-                ) for aff in institutions_ids)
+                    '1',
+                    aff
+                ) for aff in db["affiliations"].find())
 
-        # Getting the top words for each author
-        print("INFO: Creating top words for authors")
+        # Getting the top words for institution - step 2
+        if self.verbose > 1:
+            print("INFO: Creating top words for institutions - step 2")
         Parallel(
             n_jobs=self.n_jobs,
             verbose=10,
             backend="multiprocessing")(
                 delayed(top_words)(
-                    "person",
+                    '2',
+                    aff
+                ) for aff in db["affiliations"].find({"types.type": {"$in": ["faculty", "department", "group"]}}))
+
+        # Getting the top words for authors
+        if self.verbose > 1:
+            print("INFO: Creating top words for authors")
+        Parallel(
+            n_jobs=self.n_jobs,
+            verbose=10,
+            backend="multiprocessing")(
+                delayed(top_words)(
+                    '3',
                     author,
-                    "authors.id"
-                ) for author in authors_ids)
+                    self.words_inserted_ids
+                ) for author in db["person"].find({"_id": {"$nin": self.words_inserted_ids}}, no_cursor_timeout=True))
```

Comment:
the find should be,

```) for author in db["person"].find({}, no_cursor_timeout=True))```

 because if the application is restarted, words_inserted_ids will be empty anyway